### PR TITLE
 Move all clusters code to classes

### PR DESCRIPTION
**Pull request by myakove**
Depends on https://github.com/RedHatQE/cloud-tools/pull/222

### CODE REVIEW

**Comment by myakove** [comment]
Depends on https://github.com/RedHatQE/cloud-tools/pull/222

File: openshift_cli_installer/cli.py [hunk 1]
```diff
@@ -149,28 +128,30 @@
...
 )
 @click.option(
-    "--destroy-all-clusters",
+    "--destroy-clusters-from-s3-bucket",
     help="""
 \b
-Destroy all clusters under `--clusters-install-data-directory` and/or
-saved in S3 bucket (`--s3-bucket-path` `--s3-bucket-name`).
-S3 objects will be deleted upon successful deletion.
+Destroy clusters S3 bucket, --s3-bucket-name is required and optional --s3-bucket-path.
```
Comment (rnetser) [inline openshift_cli_installer/cli.py:134]:
```suggestion
Destroy clusters from S3 bucket, --s3-bucket-name is required and optional --s3-bucket-path.
```

File: openshift_cli_installer/cli.py [hunk 2]
```diff
@@ -149,28 +128,30 @@
...
     """,
+    show_default=True,
     is_flag=True,
+)
+@click.option(
+    "--destroy-clusters-from-s3-bucket-query",
```
Comment (rnetser) [inline openshift_cli_installer/cli.py:140]: readme

File: openshift_cli_installer/cli.py [hunk 3]
```diff
@@ -149,28 +128,30 @@
...
+    "--destroy-clusters-from-install-data-directory",
+    help="""
 \b
-Destroy clusters from a list of paths to `{CLUSTER_DATA_YAML_FILENAME}` files.
-The yaml file must include `s3-object-name` with s3 objet name.
-`--s3-bucket-name` and optionally `--s3-bucket-path` must be provided.
-S3 objects will be deleted upon successful deletion.
-For example:
-    '/tmp/cluster1/,/tmp/cluster2/'
+Destroy clusters from cluster data files located at --clusters-install-data-directory
```
Comment (rnetser) [inline openshift_cli_installer/cli.py:151]: example?

File: openshift_cli_installer/utils/clusters.py [hunk 4]
```diff
@@ -38,193 +22,116 @@ def get_ocm_client(ocm_token, ocm_env):
...
-            click.secho(
-                f"kubeconfig for cluster {name} does not exist; cannot run"
-                " must-gather.",
-                fg=ERROR_LOG_COLOR,
+            target_files_paths.append(target_file_path)
+
+        if download_futures:
+            for _ in as_completed(download_futures):
+                """
+                Place holder to make sure all futures are completed.
```
Comment (rnetser) [inline openshift_cli_installer/utils/clusters.py:104]: separate pr?

File: openshift_cli_installer/utils/clusters.py [hunk 5]
```diff
@@ -38,193 +22,116 @@ def get_ocm_client(ocm_token, ocm_env):
...
+
+def get_all_zip_files_from_s3_bucket(
+    client, s3_bucket_name, s3_bucket_path=None, query=None
+):
+    for _object in client.list_objects(
+        Bucket=s3_bucket_name, Prefix=s3_bucket_path
+    ).get("Contents", []):
+        _object_key = _object["Key"]
+        if _object_key.endswith(".zip"):
+            if query is None or query in _object_key:
```
Comment (rnetser) [inline openshift_cli_installer/utils/clusters.py:136]: do we want to keep it case-sensitive?

File: openshift_cli_installer/libs/clusters/ocp_clusters.py [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        for _cluster in self.clusters:
+            self.add(ocp_cluster=_cluster, **kwargs)
+
+        if self.create:
+            self.check_ocm_managed_existing_clusters()
+            self.is_region_support_hypershift()
+            self.is_region_support_aws()
+            self.is_region_support_gcp()
+
+    def add(self, ocp_cluster, **kwargs):
```
Comment (rnetser) [inline openshift_cli_installer/libs/clusters/ocp_clusters.py:49]:
```suggestion
    def add_to_cluster_lists(self, ocp_cluster, **kwargs):
```
?

File: openshift_cli_installer/libs/clusters/ocp_clusters.py [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+            self.logger.error("One cluster failed to create, destroying all clusters")
+            self.run_create_or_destroy_clusters()
+            raise click.Abort()
+
+    def attach_clusters_to_acm_cluster_hub(self):
+        for cluster in self.list_clusters:
+            if cluster.acm_clusters:
+                cluster.attach_clusters_to_acm_hub(clusters=self)
+
+    def delete_s3_target_dirs(self):
```
Comment (rnetser) [inline openshift_cli_installer/libs/clusters/ocp_clusters.py:214]: not in use

File: openshift_cli_installer/libs/clusters/ocp_clusters.py [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+        for _dir in self.s3_target_dirs:
+            shutil.rmtree(path=_dir, ignore_errors=True)
+
+    def get_cluster_by_name(self, name):
```
Comment (rnetser) [inline openshift_cli_installer/libs/clusters/ocp_clusters.py:218]:
```suggestion
    def get_cluster_obj_by_name(self, name):
```
?

**Review by rnetser — CHANGES_REQUESTED** [review]

File: openshift_cli_installer/cli.py (hunk 3 above)
Comment (myakove) [reply openshift_cli_installer/cli.py:151]: What example? this is cli flag `--clusters-install-data-directory`

File: openshift_cli_installer/utils/clusters.py (hunk 4 above)
Comment (myakove) [reply openshift_cli_installer/utils/clusters.py:104]: No

File: openshift_cli_installer/utils/clusters.py (hunk 5 above)
Comment (myakove) [reply openshift_cli_installer/utils/clusters.py:136]: Yes

File: openshift_cli_installer/cli.py (hunk 3 above)
Comment (myakove) [reply openshift_cli_installer/cli.py:152]: Update the README

**Pull request merged**